geomesa-convert/geomesa-convert-scripting/src/main/scala/org/locationtech/geomesa/convert/scripting/ScriptingFunctionFactory.scala

```diff
@@ -0,0 +1,64 @@
+/***********************************************************************
+  * Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+  * All rights reserved. This program and the accompanying materials
+  * are made available under the terms of the Apache License, Version 2.0
+  * which accompanies this distribution and is available at
+  * http://www.opensource.org/licenses/apache2.0.php.
+  *************************************************************************/
+package org.locationtech.geomesa.convert.scripting
+
+import javax.script.{Invocable, ScriptContext, ScriptEngine, ScriptEngineManager}
+
+import com.google.common.io.Files
+import org.apache.commons.io.IOUtils
+import org.locationtech.geomesa.convert.Transformers.EvaluationContext
+import org.locationtech.geomesa.convert.{TransformerFn, TransformerFunctionFactory}
+
+import scala.collection.JavaConversions._
+
+/**
+  * Provides TransformerFunctions that execute javax.scripts compatible functions defined
+  * 'classpath:geomesa-convert-scripts'.  Scripting languages are determined by the extension
+  * of the file.  For instance, 'name.js' will be interpreted as a javascript file.
+  */
+class ScriptingFunctionFactory extends TransformerFunctionFactory {
+
+  val SCRIPT_PATH = "geomesa-convert-scripts/"
+
+  lazy val functions = init()
+
+  private def init() = {
+    val loader = Thread.currentThread().getContextClassLoader
+    val scriptfiles = IOUtils.readLines(loader.getResourceAsStream(SCRIPT_PATH), "UTF-8")
```

Comment:
might be nice to have a system property to override this so we don't have to edit the jar file?

Option(System.getProperty("geomesa.convert.script.dir")).map().getOrElse...